rct229/rulesets/ashrae9012019/ruleset_functions/get_hvac_systems_primarily_serving_comp_room.py

```diff
@@ -0,0 +1,59 @@
+from rct229.rulesets.ashrae9012019.data.schema_enums import schema_enums
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_systems.baseline_system_util import (
+    find_exactly_one_zone,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_hvac_zone_list_w_area_dict import (
+    get_hvac_zone_list_w_area_by_rmi_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO, pint_sum
+
+LightingSpaceOptionsG37 = schema_enums["LightingSpaceOptions2019ASHRAE901TG37"]
+
+
+def get_hvac_systems_primarily_serving_comp_room(rmi):
+    """
+    Returns a list of HVAC systems in which greater than 50% of the area served by the HVAC system is computer room
+    space.
+
+    Parameters
+    ----------
+    rmi dict
+        A dictionary representing a ruleset model instance as defined by the ASHRAE229 schema
+
+    Returns
+    -------
+    hvac_systems_primarily_serving_comp_rooms_list list
+        A list of hvac systems in which greater than 50% of the area served by the HVAC system is computer room space.
+    """
+    hvac_zone_list_w_area_dict = get_hvac_zone_list_w_area_by_rmi_dict(rmi)
+    hvac_systems_primarily_serving_comp_rooms_list = []
+
+    for hvac in find_all(
+        "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        rmi,
+    ):
+        hvac_sys_total_floor_area = hvac_zone_list_w_area_dict[hvac["id"]]["total_area"]
+        hvac_sys_zone_id_list = hvac_zone_list_w_area_dict[hvac["id"]]["zone_list"]
+
+        hvac_zones_list = [
+            find_exactly_one_zone(rmi, zone_id) for zone_id in hvac_sys_zone_id_list
+        ]
+        hvac_spaces_list = [
+            space for zone in hvac_zones_list for space in find_all("$.spaces[*]", zone)
+        ]
+
+        hvac_system_computer_room_floor_area = pint_sum(
+            [
+                space.get("floor_area", ZERO.AREA)
+                for space in hvac_spaces_list
+                if space.get("lighting_space_type")
+                == LightingSpaceOptionsG37.COMPUTER_ROOM
+            ],
+            ZERO.AREA,
+        )
+
+        if hvac_system_computer_room_floor_area / hvac_sys_total_floor_area > 0.5:
```

Comment:
There should probably be an assertion that `hvac_sys_total_floor_area > ZERO.AREA`.

Reply:
Got it.  But I still think it best to include an assertion here as well.  It is generally dangerous to rely on the implementation of a function that is called to catch such an error.

Reply:
I agree. I wasnt sure how this can be handled since there are pre-conditions in the dependent function.
I will add the assert to make it clear here. Thanks

Reply:
BTW: it looks like I wrote the assert backwards in my comment. It should be `zone_area >= ZERO.AREA`.